modin/core/execution/ray/implementations/pandas_on_ray/io/io.py

```diff
@@ -165,6 +166,13 @@ def to_csv(cls, qc, **kwargs):
         if not cls._to_csv_check_support(kwargs):
             return RayIO.to_csv(qc, **kwargs)
 
+        if len(ray.nodes()) > 1:
```

Comment:
Can this be moved to `_to_csv_check_support`, ditto for parquet? Do we need this for `to_sql` as well?

Reply:
We could, but `_to_csv_check_support` currently just checks if we can do a parallel write, so it would be a bit of a paradigm shift to raise an error here, unless you're suggesting we return false and default to a serial write on the head node if a local path is passed?